src/main/java/uk/gov/hmcts/reform/prl/handlers/TransferToAnotherCourtEventHandler.java

```diff
@@ -0,0 +1,158 @@
+import uk.gov.hmcts.reform.prl.services.transfercase.TransferCaseContentProvider;
+import uk.gov.hmcts.reform.prl.utils.CaseUtils;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.prl.enums.YesOrNo.Yes;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class TransferToAnotherCourtEventHandler {
+    private final EmailService emailService;
+    private final TransferCaseContentProvider transferCaseContentProvider;
+
+    @EventListener(condition = "#event.typeOfEvent eq 'Transfer to another court'")
+    public void notifyAllParties(final TransferToAnotherCourtEvent event) {
+        CaseData caseData = event.getCaseData();
+
+        sendEmailToApplicantSolicitor(caseData, EmailTemplateNames.TRANSFER_COURT_EMAIL_NOTIFICATION_APPLICANT);
+
+        sendEmailToApplicant(caseData, EmailTemplateNames.TRANSFER_COURT_EMAIL_NOTIFICATION_APPLICANT);
+
+        sendEmailToRespondentSolicitor(caseData, EmailTemplateNames.TRANSFER_COURT_EMAIL_NOTIFICATION_RESPONDENT);
+
+        sendEmailToRespondents(caseData, EmailTemplateNames.TRANSFER_COURT_EMAIL_NOTIFICATION_RESPONDENT);
+
+        sendEmailToOtherParties(caseData, EmailTemplateNames.TRANSFER_COURT_EMAIL_NOTIFICATION_OTHER_PARTIES);
+    }
+
+    private void sendEmailToRespondents(CaseData caseData, EmailTemplateNames emailTemplateNames) {
+        Map<String, String> respondentEmail = caseData.getRespondents().stream()
+            .map(Element::getValue)
+            .filter(respondent -> !CaseUtils.hasLegalRepresentation(respondent)
+                && Yes.equals(respondent.getCanYouProvideEmailAddress()))
+            .collect(Collectors.toMap(
+                PartyDetails::getEmail,
+                party -> party.getFirstName() + " " + party.getLastName(),
+                (x, y) -> x
+            ));
+
+        if (!respondentEmail.isEmpty()) {
+            respondentEmail.forEach(
+                (key, value) ->
+                    emailService.send(
+                        key,
+                        emailTemplateNames,
+                        transferCaseContentProvider.buildCourtTransferEmailCitizen(caseData, value,false),
+                        LanguagePreference.getPreferenceLanguage(caseData)
+                    ));
+        }
+    }
+
+    private void sendEmailToRespondentSolicitor(CaseData caseData, EmailTemplateNames emailTemplateName) {
+        List<Map<String, List<String>>> respondentSolicitors = caseData
+            .getRespondents()
+            .stream()
+            .map(Element::getValue)
+            .filter(i -> YesNoDontKnow.yes.equals(i.getDoTheyHaveLegalRepresentation()))
+            .map(i -> {
+                Map<String, List<String>> temp = new HashMap<>();
+                temp.put(i.getSolicitorEmail(), List.of(
+                    i.getRepresentativeFirstName() + " " + i.getRepresentativeLastName(),
```

Comment:
use the common method from Party object

Reply:
used common method instead of individual method to get values